lib/network/requests/api_request.dart

```diff
@@ -0,0 +1,40 @@
+abstract class APIRequest {
+  bool authorized;
```

Comment:
You could turn this into a computed variable on the token to clean things up a bit in several places:

```suggestion
  string? token;
  bool get authorized => !!token;
```

Reply:
_Something like that_

Now all code can just care about APIRequests - but this may require a little fiddling but you get the idea (also not the most important right now)